strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/group/GroupRepositoryArtifactExistenceChecker.java

```diff
@@ -0,0 +1,98 @@
+package org.carlspring.strongbox.providers.repository.group;
+
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.providers.layout.LayoutProvider;
+import org.carlspring.strongbox.providers.layout.LayoutProviderRegistry;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.mutable.MutableBoolean;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class GroupRepositoryArtifactExistenceChecker
+{
+
+    @Inject
+    private ConfigurationManager configurationManager;
+
+    @Inject
+    private LayoutProviderRegistry layoutProviderRegistry;
+
+    public boolean artifactExistsInTheGroupRepositorySubtree(final Repository groupRepository,
```

Comment:
Cosmetic: I think there's a typo in the camel case - `artifactExistsInTheGroupRepositorySubTree`.